tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetMvc5RateLimiter.cs

```diff
@@ -0,0 +1,94 @@
+    {
+        public AspNetMvc5RateLimiterIntegratedWithSecurity(IisFixture iisFixture, ITestOutputHelper output)
+            : base(iisFixture, output, classicMode: false, enableSecurity: true, blockingEnabled: true)
+        {
+        }
+    }
+
+    [Collection("IisTests")]
+    public class AspNetMvc5RateLimiterIntegratedWithoutSecurity : AspNetMvc5RateLimiter
+    {
+        public AspNetMvc5RateLimiterIntegratedWithoutSecurity(IisFixture iisFixture, ITestOutputHelper output)
+            : base(iisFixture, output, classicMode: false, enableSecurity: false, blockingEnabled: false)
+        {
+        }
+    }
+
+    [Collection("IisTests")]
+    public class AspNetMvc5RateLimiterClassicWithSecurity : AspNetMvc5RateLimiter
+    {
+        public AspNetMvc5RateLimiterClassicWithSecurity(IisFixture iisFixture, ITestOutputHelper output)
+            : base(iisFixture, output, classicMode: true, enableSecurity: true, blockingEnabled: true)
+        {
+        }
+    }
+
+    [Collection("IisTests")]
+    public class AspNetMvc5RateLimiterClassicWithoutSecurity : AspNetMvc5RateLimiter
+    {
+        public AspNetMvc5RateLimiterClassicWithoutSecurity(IisFixture iisFixture, ITestOutputHelper output)
+            : base(iisFixture, output, classicMode: true, enableSecurity: false, blockingEnabled: false)
+        {
+        }
+    }
+
+    public abstract class AspNetMvc5RateLimiter : AspNetBase, IClassFixture<IisFixture>
+    {
+        private const int _traceRateLimit = 30;
+        private readonly IisFixture _iisFixture;
+        private readonly bool _enableSecurity;
+        private readonly bool _blockingEnabled;
+
+        public AspNetMvc5RateLimiter(IisFixture iisFixture, ITestOutputHelper output, bool classicMode, bool enableSecurity, bool blockingEnabled)
+            : base(nameof(AspNetMvc5), output, "/home/shutdown", @"test\test-applications\security\aspnet")
+        {
+            SetSecurity(enableSecurity);
+            SetAppSecBlockingEnabled(blockingEnabled);
+            _iisFixture = iisFixture;
+            _enableSecurity = enableSecurity;
+            _blockingEnabled = blockingEnabled;
+            SetEnvironmentVariable(ConfigurationKeys.AppSecTraceRateLimit, _traceRateLimit.ToString());
+
+            _iisFixture.TryStartIis(this, classicMode ? IisAppType.AspNetClassic : IisAppType.AspNetIntegrated);
+            SetHttpPort(iisFixture.HttpPort);
+        }
+
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        [Trait("LoadFromGAC", "True")]
+        [Theory]
+        [InlineData]
+        [InlineData(DefaultAttackUrl, 30)]
+        public async Task TestRateLimiterSecurity(string url = DefaultAttackUrl, int totalRequests = 80)
+        {
+            // tracing module and mvc actions
+            await TestRateLimiter(_enableSecurity, url, _iisFixture.Agent, _traceRateLimit, totalRequests, totalRequests * 2, false);
```

Comment:
One other change that would be a nice-to-have in this PR is to use the default trace rate limit so we can make sure the default configuration rate-limits to about 100

Reply:
added in 06da5c66e77807fdab8777864362407711dc0c78